File: commands/tournaments/bracket.js
const fs = require('fs');
const path = require('path');
const { DATA_PATHS, WEBSITE_REPO, GITHUB_TOKEN } = require('../../utils/constants');
const { generateBracket } = require('../../utils/bracketGen');
const { syncToSite } = require('../../utils/syncToSite');

const tournamentsPath = path.join(__dirname, '../../', DATA_PATHS.TOURNAMENTS);
const bracketPath = path.join(__dirname, '../../', DATA_PATHS.BRACKET);

module.exports = {
  name: 'bracket',
  description: 'Generate tournament bracket',

  async execute(interaction) {
    const tournamentId = interaction.options.getString('tournament_id');

    let tournaments;
    try {
      tournaments = JSON.parse(fs.readFileSync(tournamentsPath, 'utf8'));
    } catch (err) {
      console.error('❌ Failed to read tournaments.json:', err);
      return interaction.reply({ content: 'Error reading tournament data.', ephemeral: true });
    }

    const tournament = tournaments.find(t => t.id === tournamentId);
    if (!tournament) {
      return interaction.reply({ content: 'Tournament not found.', ephemeral: true });
    }

    const bracket = generateBracket(tournament.participants || []);

    try {
      fs.writeFileSync(bracketPath, JSON.stringify(bracket, null, 2));
      syncToSite('bracket.json', WEBSITE_REPO, GITHUB_TOKEN);
    } catch (err) {
      console.error('❌ Failed to write bracket.json:', err);
    }

    return interaction.reply({
      content: `Bracket for **${tournament.name}** generated!`,
      ephemeral: true
    });
  }
};

File: commands/tournaments/bracket.py
# bracket.py
import json
import discord
from discord import app_commands
from discord.ext import commands
from utils.constants import DATA_PATHS, WEBSITE_REPO, GITHUB_TOKEN
from utils.bracketGen import generate_bracket
from utils.syncToSite import sync_to_site

TOURNAMENTS_PATH = DATA_PATHS["TOURNAMENTS"]
BRACKET_PATH = DATA_PATHS["BRACKET"]

class Bracket(commands.Cog):
    def __init__(self, bot: commands.Bot):
        self.bot = bot

    @app_commands.command(
        name="bracket",
        description="Generate tournament bracket"
    )
    @app_commands.describe(tournament_id="The ID of the tournament")
    async def bracket(self, interaction: discord.Interaction, tournament_id: str):
        try:
            with open(TOURNAMENTS_PATH, "r", encoding="utf8") as f:
                tournaments = json.load(f)
        except Exception as e:
            print(f"❌ Failed to read tournaments.json: {e}")
            return await interaction.response.send_message("Error reading tournament data.", ephemeral=True)

        tournament = next((t for t in tournaments if t.get("id") == tournament_id), None)
        if not tournament:
            return await interaction.response.send_message("Tournament not found.", ephemeral=True)

        bracket = generate_bracket(tournament.get("participants", []))

        try:
            with open(BRACKET_PATH, "w", encoding="utf8") as f:
                json.dump(bracket, f, indent=2)
            sync_to_site("bracket.json", WEBSITE_REPO, GITHUB_TOKEN)
        except Exception as e:
            print(f"❌ Failed to write bracket.json: {e}")

        await interaction.response.send_message(f"Bracket for **{tournament.get('name')}** generated!", ephemeral=True)

async def setup(bot: commands.Bot):
    await bot.add_cog(Bracket(bot))
